web/server/codechecker_server/api/report_server.py

```diff
@@ -695,12 +695,19 @@ def get_comment_msg(comment):
     """
     context = webserver_context.get_context()
     message = comment.message.decode('utf-8')
-    sys_comment = comment_kind_from_thrift_type(
-        ttypes.CommentKind.SYSTEM)
+    sys_comment = comment_kind_from_thrift_type(ttypes.CommentKind.SYSTEM)
+
     if comment.kind == sys_comment:
-        elements = shlex.split(message)
-        system_comment = context.system_comment_map.get(
-            elements[0])
+        try:
+            elements = shlex.split(message)
+        except ValueError:
+            # In earlier CodeChecker we saved system comments
+            # without escaping special characters such as
+            # quotes. This is kept only for backward
+            # compatibility reason.
+            elements = shlex.split(shlex.quote(message))
```

Comment:
The result of `shlex.split()` on the result of `shlex.quote()` will always be a one-element list. This is not suitable for checking whether the first element is an indicator of system comments. So in this environment this line is equivalent with `pass`, I mean system comments stored with old CodeChecker version will not be formatted in either case. Another option would be to use simple string split, because it works both for `rev_st_changed` and `rev_st_changed_msg` but not for `comment_changed` if the comment contains spaces.